packages/react/src/components/Authenticator/shared/UserNameAlias.tsx

```diff
@@ -1,33 +1,67 @@
+import { useEffect } from 'react';
 import { I18n } from 'aws-amplify';
-import { getAliasInfoFromContext } from '@aws-amplify/ui';
+import {
+  ActorContextWithForms,
+  LoginMechanism,
+  countryDialCodes,
+  getActorContext,
+  getAliasInfoFromContext,
+} from '@aws-amplify/ui';
 
 import { useAuth } from '../../../hooks';
-import { TextField } from '../../../primitives';
+import { SelectField, TextField } from '../../../primitives';
 
 export interface UserNameAliasProps {
   handleInputChange?(event): void;
+  alias?: LoginMechanism;
   [key: string]: any;
 }
 
 export function UserNameAlias(props: UserNameAliasProps) {
-  const { handleInputChange, ...attrs } = props;
-  const [{ context }] = useAuth();
+  const { handleInputChange, alias, ...attrs } = props;
+  const [_state, send] = useAuth();
 
-  const { label, type, error } = getAliasInfoFromContext(context);
+  const actorContext: ActorContextWithForms = getActorContext(_state);
+  const { label, type, error } = getAliasInfoFromContext(_state.context, alias);
   const i18nLabel = I18n.get(label);
 
+  const isPhoneAlias = type === 'tel';
+
+  useEffect(() => {
+    isPhoneAlias &&
+      send({
+        type: 'CHANGE',
+        data: { name: 'country_code', value: actorContext.country_code },
+      });
+  }, []);
```

Comment:
On init, update xstate with the default country code when the alias is a phone number.